backend/api/views.py

```
@@ -211,6 +342,306 @@ def perform_destroy(self, instance):
     queryset = WaypointMedia.objects.all()
     serializer_class = WaypointMediaSerializer
+
+    def get_queryset(self):
+        return WaypointMedia.objects.filter(
+            waypoint__group__activity__in=get_accessible_activity_queryset(self.request.user)
+        )
+
+    def perform_create(self, serializer):
+        waypoint_id = self.request.data.get("waypoint")
+        if not waypoint_id:
+            raise ValidationError("Missing waypoint")
```

Ruff (0.15.2)

[warning] Avoid specifying long messages outside the exception class

(TRY003)